### ISSUE TITLE
Issues with ls(...) filtering for a mid string wildcard (text prefix and trailer exact match though)

### ISSUE THREAD

**Issue by OneOfMany07** [bitburner-official/bitburner-src#400]
I'm trying to get only files that match /Temp/*.txt and can't seem to manage with the current ls(...) API.

https://github.com/bitburner-official/bitburner-src/blob/dev/markdown/bitburner.ns.ls.md

Also the variable is named "grep" which accepts Regular Expressions, but the help information says "substring".  I could use RE to get the functionality I want ('/Temp/.+\.txt').  But I don't think substring would work as there aren't wildcards in substrings, right?

I'd suggest naming the parameter if you plan to keep it as substring.  Or change the feature to support regular expressions somehow like grep does.

**Comment by Snarling** [comment]
Since you are in js and not a terminal, you can filter the output yourself using regex, e.g.
```js
let tempTextFiles = ns.ls("home").filter(filename=>filename.match(/^\/Temp\/.*\.txt$/))
```
Implementing any sort of 1:1 correspondence with grep is unlikely, so I agree that the grep parameter should be renamed.

**Comment by Snarling** [comment]
Fixed by c2cd584 (Just documenting further that it's only a substring).
